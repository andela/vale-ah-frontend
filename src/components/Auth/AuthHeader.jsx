import React from 'react';
import PropTypes from 'prop-types';
import SocialButton from '../SocialButton/SocialButton';

/**
 * Auth Component
 * @param {object} props
 * @returns {JSX.Element} Auth component
 */
const AuthHeader = ({ authType }) => {
  const [titleText, authTypeText] =
    authType === 'login' ? ['Welcome to', 'Sign in'] : ['Join', 'Sign up'];
  return (
    <header>
      <h1 className="title">{titleText} Naija ChopChop</h1>
      <SocialButton authType={authTypeText} provider="Facebook" />
      <SocialButton authType={authTypeText} provider="Google" />
      <SocialButton authType={authTypeText} provider="Twitter" />
    </header>
  );
};

AuthHeader.propTypes = {
  authType: PropTypes.string.isRequired,
};

export default AuthHeader;
