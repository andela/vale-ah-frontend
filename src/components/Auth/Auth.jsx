import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import Button from '../Button/Button';
import AuthHeader from './AuthHeader';

/**
 * Auth component
 * @param {object} props properties
 * @returns {JSX.Element} Auth component
 */
const Auth = ({ authType, onClose }) => {
  let submitButtonText;
  let otherAuthText;

  if (authType === 'login') {
    submitButtonText = 'Log in';
    otherAuthText = 'Create account';
  } else if (authType === 'register') {
    submitButtonText = 'Sign up';
    otherAuthText = 'Log in';
  }

  return (
    <div className="auth">
      <div className="close-button">
        <button type="button" onClick={onClose}>
          Ã—
        </button>
      </div>
      <form>
        <AuthHeader authType={authType} />
        <main>
          <fieldset>
            <legend>or</legend>
            {authType === 'register' && (
              <input
                className="text-field"
                type="text"
                name="username"
                placeholder="Username"
              />
            )}

            <input
              className="text-field"
              type="email"
              name="email"
              placeholder="Email"
            />
            <input
              className="text-field"
              type="password"
              name="password"
              placeholder="Password"
            />
            <p className="forgot-password">Forgot password?</p>
            <Button type="submit" color="green">
              {submitButtonText}
            </Button>
          </fieldset>
        </main>
      </form>
      <p className="other-auth">
        {`You ${authType === 'login' ? "don't" : 'already'} have an account? `}
        <Link
          to={`/${authType === 'login' ? 'register' : 'login'}`}
          className="action"
        >
          {otherAuthText}
        </Link>
      </p>
    </div>
  );
};

Auth.propTypes = {
  authType: PropTypes.string.isRequired,
  onClose: PropTypes.func.isRequired,
};

export default Auth;
