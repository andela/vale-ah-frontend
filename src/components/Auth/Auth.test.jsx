import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Auth from './Auth';

configure({ adapter: new Adapter() });
describe('<Auth />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<Auth authType="login" />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
