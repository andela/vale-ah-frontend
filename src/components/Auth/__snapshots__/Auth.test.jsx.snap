// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Auth /> matches the snapshot 1`] = `
<div
  className="auth"
>
  <div
    className="close-button"
  >
    <button
      type="button"
    >
      Ã—
    </button>
  </div>
  <form>
    <AuthHeader
      authType="login"
    />
    <main>
      <fieldset>
        <legend>
          or
        </legend>
        <input
          className="text-field"
          name="email"
          placeholder="Email"
          type="email"
        />
        <input
          className="text-field"
          name="password"
          placeholder="Password"
          type="password"
        />
        <p
          className="forgot-password"
        >
          Forgot password?
        </p>
        <Button
          color="green"
          onClick={[Function]}
          type="submit"
        >
          Log in
        </Button>
      </fieldset>
    </main>
  </form>
  <p
    className="other-auth"
  >
    You don't have an account? 
    <Link
      className="action"
      replace={false}
      to="/register"
    >
      Create account
    </Link>
  </p>
</div>
`;
